### PR TITLE
Feature: Realm Message of the Day

### PR DESCRIPTION
**Pull request by kirmmin**
**Features**
- Realm Message of the Day displayed to players each time they login
- Message of the Day live updating support
   - This includes a modification to `ConfigurationManager` & `SharedConfiguration` classes allowing for it to save changes to the `WorldServer.json` configuration on MOTD changes.

### CODE REVIEW

File: Source/NexusForever.Shared/Configuration/SharedConfiguration.cs [hunk 1]
```diff
@@ -14,17 +14,21 @@ public static class SharedConfiguration
         private static readonly ILogger log = LogManager.GetCurrentClassLogger();
         public static IConfiguration Configuration { get; private set; }
 
+        private static string fileLocation { get; set; }
```
Comment (Fabi) [inline Source/NexusForever.Shared/Configuration/SharedConfiguration.cs]: Don't use ` { get; set; }` on private fields :P

File: Source/NexusForever.Shared/Configuration/SharedConfiguration.cs [hunk 2]
```diff
@@ -100,5 +104,11 @@ private static void ConvertSingleDatabaseSection(string path, JObject root)
             selectedToken.Add("ConnectionString", JValue.CreateString(connectionString));
             selectedToken.Add("Provider", JValue.CreateString(DatabaseProvider.MySql.ToString()));
         }
+
+        public static void Save<T>(object config)
+        {
+            if (fileLocation.Length > 0)
+                File.WriteAllText(fileLocation, JsonConvert.SerializeObject(config, Formatting.Indented));
```
Comment (Fabi) [inline Source/NexusForever.Shared/Configuration/SharedConfiguration.cs]: What if another process or the same process blocks file writing somehow?

**Comment by kirmmin** [comment]
Rebased today to the latest `develop` branch. Also, applied some protections for the Save method failing.

**Comment by kirmmin** [comment]
Rebased and update today.

**Pull request merged**